matchms/filtering/metadata_processing/repair_smiles_of_salts.py

```diff
@@ -17,6 +19,9 @@ def repair_smiles_of_salts(spectrum_in,
     spectrum = spectrum_in.clone()
 
     smiles = spectrum.get("smiles")
+    if not is_valid_smiles(smiles):
+        logger.warning("No valid smiles, please run require_valid_annotation before")
```

Comment:
I think this is too focused on workflows on (partly-) annotated data.
In many cases people will run default pipelines for data that is largely not annotated, in which case they will be flooded with warnings :)

Since we also offer no filter tree structure (such as "if valid smiles run this filter..."), we should be less strict here and also accept that there is simply no smiles --> `if smiles is not None and not is_valid_smiles(smiles):`

Reply:
Or, better I think to add:
```python
if smiles is None:
    return spectrum
```

Reply:
@niekdejonge I added this bit of code. Hope that works.